cpp/include/cudf/detail/gather.cuh

```diff
@@ -188,6 +190,59 @@ struct column_gatherer_impl<string_view, MapItType>
 
 };
 
+ /**
+  * @brief Column gather specialization for dictionary column type.
+  */
+template<typename MapItType>
+struct column_gatherer_impl<dictionary32, MapItType>
+{
+ /**
+  * @brief Type-dispatched function to gather from one column to another based
+  * on a `gather_map`.
+  *
+  * @param source_column View into the column to gather from
+  * @param gather_map_begin Beginning of iterator range of integral values representing the gather map
+  * @param gather_map_end End of iterator range of integral values representing the gather map
+  * @param nullify_out_of_bounds Nullify values in `gather_map` that are out of bounds
+  * @param mr Memory resource to use for all allocations
+  * @param stream CUDA stream on which to execute kernels
+  * @return New dictionary column with gathered rows.
+  */
+  std::unique_ptr<column> operator()(column_view const& source_column,
+                                     MapItType gather_map_begin,
+                                     MapItType gather_map_end,
+                                     bool nullify_out_of_bounds,
+                                     rmm::mr::device_memory_resource *mr,
+                                     cudaStream_t stream) {
+      dictionary_column_view dictionary(source_column);
+      auto output_count = std::distance(gather_map_begin, gather_map_end);
+      if( output_count == 0 )
+          return make_empty_column(data_type{DICTIONARY32});
+      // we decided we will copy the keys for gather
```

Comment:
Rather than writing "we  decided", perhaps just add a note about this in the documentation for gather. Here, you could document possible alternatives?